### PR TITLE
feat: [CI-6878]: Adding `python` & `powershell` support for syntax highlighting

### PR DESCRIPTION
**Pull request by vardanbansal-harness**
### Summary

Sample for Python:

image

Sample for Powershell:

image

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-ci/components/PipelineSteps/RunStep/RunStepBase.tsx [hunk 1]
```diff
@@ -121,8 +135,8 @@ export const RunStepBase = (
     >
       {(formik: FormikProps<RunStepData>) => {
         // This is required
+        const scriptType = getScriptTypeForShell(formik)
```
Comment (abhinavrastogi-harness) [inline src/modules/75-ci/components/PipelineSteps/RunStep/RunStepBase.tsx:138]: nit: the comment applies to the line below (for setFormikRef). let''s not break that coupling.

**Review by abhinavrastogi-harness — COMMENTED** [review]
have you evaluated what is the impact on the bundle? is the python parser loaded on demand, or is it getting bundled with existing monaco? what is the cost of the new parser?
impact on load time?

**Comment by abhinavrastogi-harness** [comment]
Also, shouldn't this be behind a feature flag?
I am assuming it requires corresponding support on the backend as well to understand python.

File: src/modules/75-ci/components/PipelineSteps/RunStep/RunStepBase.tsx (hunk 1 above)
Comment (vardanbansal-harness) [reply src/modules/75-ci/components/PipelineSteps/RunStep/RunStepBase.tsx]: Fixed it, thanks for pointing that out 👍🏻

**Comment by vardanbansal-harness** [comment]
> Also, shouldn't this be behind a feature flag? I am assuming it requires corresponding support on the backend as well to understand python.

Backend support was already added in an earlier sprint.

**Comment by vardanbansal-harness** [comment]
> have you evaluated what is the impact on the bundle? is the python parser loaded on demand, or is it getting bundled with existing monaco? what is the cost of the new parser? impact on load time?

Let me do this analysis and get back to you.

**Comment by vardanbansal-harness** [comment]
> > have you evaluated what is the impact on the bundle? is the python parser loaded on demand, or is it getting bundled with existing monaco? what is the cost of the new parser? impact on load time?
> 
> Let me do this analysis and get back to you.

@abhinavrastogi-harness I did a side-by-side comparison of `yarn build` for this branch vs develop and do not see any change in bundle size as a result of this change. 

Screen Shot 2023-02-23 at 6 12 15 PM

Screen Shot 2023-02-23 at 6 12 03 PM

**Comment by vardanbansal-harness** [comment]
> > > have you evaluated what is the impact on the bundle? is the python parser loaded on demand, or is it getting bundled with existing monaco? what is the cost of the new parser? impact on load time?
> > 
> > 
> > Let me do this analysis and get back to you.
> 
> @abhinavrastogi-harness I did a side-by-side comparison of `yarn build` for this branch (left) vs develop(right) and do not see any change in bundle size as a result of this change.
> 
> Screen Shot 2023-02-23 at 6 12 03 PM
> 
> Screen Shot 2023-02-23 at 6 12 15 PM

Also did a performance profiling for the user selection `Python` from the `Shell` dropdown, as a result of python parser is getting loaded on demand. 

image

**Review by shaurya-harness — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]
LGTM. Thanks for the performance analysis!

**Pull request merged**